src/main/java/org/opentripplanner/middleware/connecteddataplatform/AnonymizedTripRequest.java

```diff
@@ -50,24 +70,122 @@ public AnonymizedTripRequest() {
 
     public AnonymizedTripRequest(
         TripRequest tripRequest,
+        FindIterable<TripSummary> tripSummaries,
         Coordinates fromCoordinates,
         Coordinates toCoordinates
     ) {
         this.batchId = tripRequest.batchId;
         this.fromPlace = fromCoordinates;
         this.toPlace = toCoordinates;
-        if (tripRequest. requestParameters != null) {
+        if (tripRequest.requestParameters != null) {
             this.date = tripRequest.requestParameters.get("date");
             this.time = tripRequest.requestParameters.get("time");
-            String tripType = tripRequest.requestParameters.get("arriveBy");
-            if (tripType != null && tripType.equalsIgnoreCase("true")) {
+            String tripRequestType = tripRequest.requestParameters.get("arriveBy");
+            if (tripRequestType != null && tripRequestType.equalsIgnoreCase("true")) {
                 this.tripType = AnonymousTripType.ARRIVE_BY;
-            } else if (tripType != null && tripType.equalsIgnoreCase("false")) {
+            } else if (tripRequestType != null && tripRequestType.equalsIgnoreCase("false")) {
                 this.tripType = AnonymousTripType.DEPART_AT;
             }
-            this.mode = tripRequest.requestParameters.get("mode");
+            this.mode = getModes(tripRequest.requestParameters.get("mode"));
             this.maxWalkDistance = tripRequest.requestParameters.get("maxWalkDistance");
             this.optimize = tripRequest.requestParameters.get("optimize");
         }
+
+        // Extract all trip summary itineraries, convert to anonymized itineraries and group.
+        int tripSummaryId = 1;
+        for (TripSummary tripSummary : tripSummaries) {
+            if (tripSummary.error != null) {
+                // If trip summary has an error, add it to the anonymized trip request and don't attempt to process the
+                // trip summary itineraries, because there won't be any.
+                this.error = tripSummary.error;
+                break;
```

Comment:
My take on this, and it could be wrong, is that if one of the trip summaries has an error they all will. So, if the first trip summary has an error, record this and then don't process the itineraries or subsequent trip summaries. In this case`tripSummaryId` is redundant.

Reply:
Oh yeah this makes sense thanks!